### PR TITLE
Wrapping cmake macro simplification

### PR DESCRIPTION
**Pull request by hjmjohnson**
Python wrapping code is simplified (inline single use functions) and reduce contamination in the global namespace.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Review by thewtex — APPROVED** [review]
LGTM :-)

We can refactor our way into cleanliness :-) :1234: 

Please make sure to test locally with `ITK_WRAP_DOC` enabled before merging.

File: Wrapping/Generators/Doc/CMakeLists.txt [hunk 1]
```diff
@@ -109,4 +109,8 @@ macro(itk_end_wrap_submodule_DOC module_name)
   #    COMMENT "-- Wrapping library ${module_name}: Generating swig interface for inline documentation."
     )
     list(APPEND ITK_WRAP_DOC_DOCSTRING_FILES ${swig_doc_interface_file})
+
+    unset(swig_doc_interface_file)
```
Comment (dzenanz) [inline Wrapping/Generators/Doc/CMakeLists.txt:97]: A few of these `unset` commits could be squashed together.

File: Wrapping/Generators/CMakeLists.txt [hunk 2]
```diff
@@ -149,10 +149,10 @@ foreach(lang ${generators})
   # add the language to the list, if the user have turn it on, and include
   # the subdirectory
   if(${module_prefix}_WRAP_${lang_upper})
-    set(WRAP_ITK_GENERATORS ${WRAP_ITK_GENERATORS} "${lang}")
+    list(APPEND WRAP_ITK_GENERATORS "${lang}")
```
Comment (dzenanz) [inline Wrapping/Generators/CMakeLists.txt:71]: squash with previous

File: Wrapping/Generators/CastXML/CMakeLists.txt [hunk 3]
```diff
@@ -171,15 +171,6 @@ mark_as_advanced(CASTXML_EXECUTABLE)
 set(ITK_WRAP_CASTXML_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}" CACHE INTERNAL "castxml source dir")
 
 
-macro(itk_wrap_include_castxml include_file)
```
Comment (dzenanz) [inline Wrapping/Generators/CastXML/CMakeLists.txt:174]: Perhaps either squash all `inline` commits together, or separate this one into two commits.

**Review by dzenanz — APPROVED** [review]
Mostly looks good. Some commit squashing and splitting might be in order.

File: Wrapping/Generators/Doc/CMakeLists.txt (hunk 1 above)
Comment (hjmjohnson) [reply Wrapping/Generators/Doc/CMakeLists.txt:97]: Done.

File: Wrapping/Generators/CMakeLists.txt (hunk 2 above)
Comment (hjmjohnson) [reply Wrapping/Generators/CMakeLists.txt:71]: Done.

**Comment by hjmjohnson** [comment]
@thewtex   ITK_WRAP_DOC:BOOL=ON succeeded (after doxygen and graphviz installs made).

```txt
Generating docs for compound itk::Statistics::RandomVariateGeneratorBase...
Generating graph info page...
Generating directory documentation...
Generating dependency graph for directory Core
Generating dependency graph for directory include
Generating dependency graph for directory Mesh
Generating dependency graph for directory Modules
Generating index page...
Generating page index...
Generating module index...
Generating namespace index...
Generating namespace member index...
Generating annotated compound index...
Generating alphabetical compound index...
Generating hierarchical class index...
Generating member index...
Generating file index...
Generating file member index...
Generating example index...
finalizing index lists...
writing tag file...
Generating search index
Running plantuml with JAVA...
Running dot...
Patching output file 1/1
lookup cache used 7517/262144 hits=58690 misses=8020
finished...
[2144/2144] Linking CXX shared module Wrapping/Generators/Python/itk/_ITKMetricsv4Python.so

Build finished
```

@dzenanz Squashed many like topics together.  Did not change the codebase.

**Pull request merged**